lib/dialects/oracle/data-types.js

```diff
@@ -0,0 +1,397 @@
+  BaseTypes.DATEONLY.types.oracle = ['DATE', 'DATEONLY'];
+  BaseTypes.BOOLEAN.types.oracle = ['NUMBER'];
+  BaseTypes.BLOB.types.oracle = ['BLOB'];
+  BaseTypes.DECIMAL.types.oracle = ['DECIMAL'];
+  BaseTypes.UUID.types.oracle = false;
+  BaseTypes.ENUM.types.oracle = false;
+  BaseTypes.REAL.types.oracle = false;
+  BaseTypes.NUMERIC.types.oracle = false;
+  BaseTypes.DOUBLE.types.oracle = false;
+  // BaseTypes.GEOMETRY.types.oracle = [240]; // not yet supported
+  BaseTypes.GEOMETRY.types.oracle = false;
+
+
+  function BLOB(length) {
+    if (!(this instanceof BLOB)) return new BLOB(length);
+    BaseTypes.BLOB.apply(this, arguments);
+  }
+  inherits(BLOB, BaseTypes.BLOB);
+
+  BLOB.prototype.toSql = function toSql() {
+    if (this._length) {
+      if (this._length.toLowerCase() === 'tiny') { // tiny = 2^8
+        warn('ORACLE does not support BLOB with the `length` = `tiny` option. `RAW(256)` will be used instead.');
+        return 'RAW(256)';
+      }
+      warn('ORACLE does not support BLOB with the `length` option. `RAW(2000)` will be used instead.');
+      if (isNaN(this.length) || this.length > 2000) {
+        return 'RAW(2000)';
+      } else {
+        return `RAW(${this._length})`;
+      }
+    }
+    return 'BLOB';
+  };
+
+  BLOB.prototype._hexify = function _hexify(hex) {
+    return 'hextoraw(\'' + hex + '\')';
+  };
+
+  function CHAR(length) {
+    if (!(this instanceof CHAR)) return new CHAR(length);
+    BaseTypes.CHAR.apply(this, arguments);
+  }
+  inherits(CHAR, BaseTypes.CHAR);
+
+  CHAR.prototype.toSql = function toSql() {
+    if (this._binary) {
+      return 'RAW(' + this._length + ')';
+    }
+    return BaseTypes.CHAR.prototype.toSql.call(this);
+  };
+
+  function STRING(length, binary) {
+    if (!(this instanceof STRING)) return new STRING(length, binary);
+    BaseTypes.STRING.apply(this, arguments);
+  }
+  inherits(STRING, BaseTypes.STRING);
+
+  STRING.prototype.toSql = function toSql() {
+    if (this.length > 4000 || this._binary && this._length > 2000) {
+      warn('Oracle 12 supports length up to 32764; be sure that your administrator has extended the MAX_STRING_SIZE parameter. Check https://docs.oracle.com/database/121/REFRN/GUID-D424D23B-0933-425F-BC69-9C0E6724693C.htm#REFRN10321');
+    }
+    if (!this._binary) {
+      return 'NVARCHAR2(' + this._length + ')';
+    } else {
+      return 'RAW(' + this._length + ')';
+    }
+  };
+
+  STRING.prototype.escape = false;
+
+  STRING.prototype._stringify = function _stringify(value, options) {
+    if (this._binary) {
+      return BLOB.prototype._stringify(value);
+    } else {
+      return options.escape(value);
+    }
+  };
+
+  function TEXT(length) {
+    if (!(this instanceof TEXT)) return new TEXT(length);
+    BaseTypes.TEXT.apply(this, arguments);
+  }
+  inherits(TEXT, BaseTypes.TEXT);
+
+  TEXT.prototype.toSql = function toSql() {
+    //TEXT is not support by Oracle, passing through NVARCHAR
```

Comment:
NVARCHAR2 is a specialized type with a particular character set; I don't know what TEXT is but from the name wouldn't it map to VARCHAR2?

Also, there is no support for NVARCHAR2 in node-oracledb v1.  It may work in some cases where the DB charset width is big enough to hold the DB national character set.  (Node-oracledb v2 has query support for NVARCHAR2)

Reply:
TEXT is an extended varchar type in mssql (https://docs.microsoft.com/en-us/sql/t-sql/data-types/ntext-text-and-image-transact-sql) except if I understood wrong.

That's why I've decided to map it to NVARCHAR2. But if no length is passed, I set it to CLOB

Reply:
What do you map NTEXT to?